### PR TITLE
fix(helm): add warnings for missing chart dependencies

### LINKED ISSUE #1567: Issue warning when requirements.yaml has something that charts/ doesn't?
Helm will happily install a chart without the requirements being available in charts/. Please issue a warning when that is the case.

### PR DESCRIPTION
**Pull request by wilkers-steve**
When 'helm install', 'helm package', and 'helm upgrade' are run,
Helm will not issue any warnings if any dependencies listed in
a chart's requirements.yaml file are missing.  This change includes
warnings when a chart is found in requirements.yaml but isn't
in charts/.

The next logical step is to include a `--strict` flag to cancel the operation if dependencies are missing.  As this is a feature required for 2.2, I wanted to make sure this PR was submitted to make the release cut.

Closes #1567

### CODE REVIEW

**Comment by wilkers-steve** [comment]
I signed it!

File: cmd/helm/install.go [hunk 1]
```diff
@@ -388,3 +397,19 @@ func defaultNamespace() string {
 	}
 	return "default"
 }
+
+func (i *installCmd) checkDependencies(ch *chart.Chart, reqs *chartutil.Requirements) {
+	deps := ch.GetDependencies()
```
Comment (technosophos) [inline cmd/helm/install.go]: If you created a function `func checkDependencies(ch *chart.Chart, reqs *chartutil.Requirements, out io.Writer)`, could you combine the three different versions of this function?

**Review by technosophos — COMMENTED** [review]
Other than wondering if we can combine the three similar functions into one, the rest looks great.

File: cmd/helm/install.go (hunk 1 above)
Comment (wilkers-steve) [inline cmd/helm/install.go]: @technosophos Yeah, that's totally doable.  Let me throw that together real quick

**Comment by thomastaylor312** [comment]
This also looks good to me and I agree with @technosophos comment. Let us know when you've pushed up the new combined and one of us can test it!

**Comment by wilkers-steve** [comment]
@technosophos @thomastaylor312, I've pushed up the most recent change for combining the 3 functions.

**Comment by technosophos** [comment]
Merging now. Thanks!

**Pull request merged**